tests/samplers_tests/test_samplers.py

```diff
@@ -229,6 +242,142 @@ def sample() -> float:
     np.testing.assert_almost_equal(round_points, points)
 
 
+@parametrize_relative_sampler
+@pytest.mark.parametrize(
+    "x_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+@pytest.mark.parametrize(
+    "y_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+def test_sample_relative_numerical(
+    relative_sampler_class: Callable[[], BaseSampler],
+    x_distribution: BaseDistribution,
+    y_distribution: BaseDistribution,
+) -> None:
+
+    search_space: Dict[str, BaseDistribution] = OrderedDict(x=x_distribution, y=y_distribution)
+    study = optuna.study.create_study(sampler=relative_sampler_class())
+    trial = study.ask(search_space)
+    study.tell(trial, sum(trial.params.values()))
+
+    def sample() -> List[float]:
+        params = study.sampler.sample_relative(study, _create_new_trial(study), search_space)
+        return [params[name] for name in search_space]
+
+    points = np.array([sample() for _ in range(10)])
+    for i, distribution in enumerate(search_space.values()):
+        assert isinstance(
+            distribution,
+            (
+                UniformDistribution,
+                LogUniformDistribution,
+                DiscreteUniformDistribution,
+                IntUniformDistribution,
+                IntLogUniformDistribution,
+            ),
+        )
+        assert np.all(points[:, i] >= distribution.low)
+        assert np.all(points[:, i] <= distribution.high)
```

Comment:
To be exact, we should change `<=` to `<` for uniform and loguniform, but I also think this is acceptable.

Reply:
Thank you for your comment. #2223 changed the range definitions of `UniformDistribution` and `LogUniformDistribution`, and I use `<=` for those. (Honestly speaking, the definitions are different from that of `suggest_float` and it is confusing. We may want to revise it in the version 3 discussion.)

I checked the distribution reference and the current specification of the ranges is as follows:

- [`UniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.UniformDistribution.html#optuna.distributions.UniformDistribution): `[low , high]` (Both included)
- [`LogUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.LogUniformDistribution.html#optuna.distributions.LogUniformDistribution): `[low, high]`
- [`DiscreteUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.DiscreteUniformDistribution.html#optuna.distributions.DiscreteUniformDistribution): `[low, high]`
- [`IntUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.IntUniformDistribution.html#optuna.distributions.IntUniformDistribution): `[low, high]`
- [`IntLogUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.IntLogUniformDistribution.html#optuna.distributions.IntLogUniformDistribution): `[low, high]`